### PR TITLE
Don't query interactive layer ids before map has loaded

### PR DESCRIPTION
**Pull request by macobo**
This can cause an false-positive error such as `The layer 'foobar' does
not exist in the maps style and cannot be queries for features`.

This needs tests, though the author of this PR doesn't know how to
create these without enzyme.

Solves #707

### CODE REVIEW

File: src/components/interactive-map.js [hunk 1]
```diff
@@ -347,6 +350,11 @@ export default class InteractiveMap extends PureComponent<InteractiveMapProps, S
     return event;
   }
 
+  _onLoad = () => {
+    this.setState({stylesLoaded: true});
+    this.props.onLoad();
```
Comment (Pessimistress) [inline src/components/interactive-map.js]: Need to forward the event object to the callback.

File: src/components/interactive-map.js [hunk 2]
```diff
@@ -204,6 +205,8 @@ export default class InteractiveMap extends PureComponent<InteractiveMapProps, S
   }
 
   state : State = {
+    // Whether mapbox styles have finished loading
+    stylesLoaded: false,
```
Comment (Pessimistress) [inline src/components/interactive-map.js]: Just call it `isLoaded`?

File: src/components/interactive-map.js (hunk 1 above)
Comment (macobo) [reply src/components/interactive-map.js]: I couldn't find documentation what the event object here is, but you're right that there is an event being triggered here.

Interestingly enough, the [typescript type definitions here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-map-gl/index.d.ts#L43) also aren't aware there's an event object.

Will fix though, thanks for catching this!

**Comment by macobo** [comment]
Thanks for the review! I think this is ready for another round @Pessimistress

File: src/components/interactive-map.js [hunk 3]
```diff
@@ -390,7 +398,7 @@ export default class InteractiveMap extends PureComponent<InteractiveMapProps, S
     if (!this.state.isDragging) {
       const {onHover, interactiveLayerIds} = this.props;
       let features;
-      if (interactiveLayerIds || onHover) {
+      if (this.state.isLoaded && (interactiveLayerIds || onHover)) {
```
[the remark this answers is not in the document]
Comment (macobo) [reply src/components/interactive-map.js]: Thanks. Opted to always normalize the event.

**Comment by Pessimistress** [comment]
Thank you for contributing!

(Looks like CI is failing for unrelated reasons)

**Comment by macobo** [comment]
Can confirm, `npm test` runs without fail on local machine.

Thanks for the quick reviews @Pessimistress. Feel free to close any relevant issues now.

This library is such pleasure to use. Thanks for building and maintaining it! Now if only we could figure out proper drawing controls at my workplace we could get rid of google maps for good...

**Pull request merged**